app/src/main/java/com/simplymadeapps/preferencehelper/PreferenceHelper.java

```diff
@@ -31,25 +45,43 @@ public static void init(Context context) {
             throw new IllegalArgumentException("Key cannot be null");
         }
 
-        if(value == null || value instanceof String) {
-            // We assume a null value is a string because the other primitive types cannot be null and we will not be storing null sets
+        if(value == null && type == null) {
+            throw new IllegalArgumentException("You must specify the stored object type when storing a null value using put(String key, T value, Class<T> type)");
```

Comment:
Move these IllegalArgumentException to a private method.

Reply:
I ended up moving all exception checks to one place because it felt right.